test/test-harness.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+export SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+set -euo pipefail
+
+TEST_CASE_SCRIPT=""
+
+USAGE=$(cat << 'EOM'
+  Usage: test-harness.sh  [-c <case script path>]
+  Executes the test harness for the EKS Optimized AL2 AMI.
+  By default the test harness executes all scripts in the cases directory.
+  Example: test-harness.sh
+          Optional:
+            -c          A path to a specific test case script
+EOM
+)
+
+while getopts "c:h" opt; do
+  case ${opt} in
+    c ) # Case Script Path
+        TEST_CASE_SCRIPT="$OPTARG"
+      ;;
+    h ) # help
+        echo "$USAGE" 1>&2
+        exit
+      ;;
+    \? )
+        echo "$USAGE" 1>&2
+        exit
+      ;;
+  esac
+done
+
+docker build -t eks-optimized-ami -f "${SCRIPTPATH}/Dockerfile" "${SCRIPTPATH}/../"
+overall_status=0
+
+function run(){
+    local temp_dir=$1
+    shift
+    cp -f ${SCRIPTPATH}/../files/kubelet-config.json ${temp_dir}/kubelet-config.json
+    docker run -v ${SCRIPTPATH}/../files/bootstrap.sh:/etc/eks/bootstrap.sh \
+        -v ${SCRIPTPATH}/../files/max-pods-calculator.sh:/etc/eks/max-pods-calculator.sh \
+        -v ${temp_dir}/kubelet-config.json:/etc/kubernetes/kubelet/kubelet-config.json \
+        -it eks-optimized-ami $@
```

Comment:
is volume-mounting the entire `files/` dir a concern?

also can we add `--rm` to clean up the container?